### PR TITLE
Remove token computer dependency from transactions factories & compute bytes to verify transaction signature

### PR DESCRIPTION
**Pull request by popenta**
This PR introduces a few **breaking changes**. These changes we're introduced to match the specs and the `sdk-js-core` implementation.

The `SmartContractTransactionsFactory` and `TransferTransactionsFactory` don't need to be instantiated using an object of type `token_computer: ITokenComputer`.

The `IAddressConverter` interface was removed as it was not used.

### CODE REVIEW

File: multiversx_sdk/core/transactions_factories/smart_contract_transactions_factory.py [hunk 1]
```diff
@@ -23,19 +24,11 @@ class IConfig(Protocol):
...
-    def is_fungible(self, token: IToken) -> bool:
-        ...
-
-    def extract_identifier_from_extended_identifier(self, identifier: str) -> str:
-        ...
-
-
 class SmartContractTransactionsFactory:
-    def __init__(self, config: IConfig, token_computer: ITokenComputer) -> None:
+    def __init__(self, config: IConfig) -> None:
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transactions_factories/smart_contract_transactions_factory.py:28]: We should document this breaking change.

File: multiversx_sdk/core/transactions_factories/transfer_transactions_factory.py [hunk 2]
```diff
@@ -21,19 +22,11 @@ class IConfig(Protocol):
...
-    def is_fungible(self, token: IToken) -> bool:
-        ...
-
-    def extract_identifier_from_extended_identifier(self, identifier: str) -> str:
-        ...
-
-
 class TransferTransactionsFactory:
-    def __init__(self, config: IConfig, token_computer: ITokenComputer) -> None:
+    def __init__(self, config: IConfig) -> None:
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transactions_factories/transfer_transactions_factory.py:26]: We should document this breaking change.

File: multiversx_sdk/core/transaction_computer.py [hunk 3]
```diff
@@ -41,6 +41,14 @@ def compute_bytes_for_signing(self, transaction: ITransaction) -> bytes:
         serialized = self._dict_to_json(dictionary)
         return serialized
 
+    def compute_bytes_for_verifying(self, transaction: ITransaction) -> bytes:
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction_computer.py:44]: For reviewers: https://github.com/multiversx/mx-sdk-specs/pull/62

File: multiversx_sdk/core/transaction.py [hunk 4]
```diff
@@ -1,22 +1,10 @@
-from typing import Optional, Protocol
+from typing import Optional
 
-from multiversx_sdk.core.constants import (DEFAULT_HRP,
-                                           TRANSACTION_MIN_GAS_PRICE,
+from multiversx_sdk.core.constants import (TRANSACTION_MIN_GAS_PRICE,
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction.py:3]: So the `IAddressConverter` was never used? Though, we can mention this removal as a breaking change, perhaps, as well.

File: multiversx_sdk/core/transaction_test.py [hunk 5]
```diff
@@ -292,3 +293,55 @@ def test_ensure_transaction_is_valid(self):
         self.transaction_computer.apply_options_for_hash_signing(tx)
         assert tx.version == 2
         assert tx.options == 3
+
+    def test_compute_bytes_for_verifying_signature(self):
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction_test.py:297]: :+1: 

https://github.com/multiversx/mx-sdk-js-core/pull/432

**Review by andreibancioiu — APPROVED** [review]
We should document the changes in the PR description, then in the release notes (at the time of release).

File: multiversx_sdk/core/transaction.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk/core/transaction.py:3]: It was not used. Added in the PR description.

**Review by ssd04 — APPROVED** [review]

**Pull request merged**